### ISSUE TITLE
`computeAccessibleName` won't compute combobox name from self referencing aria-labelledby attr

### ISSUE THREAD

**Issue by agentdylan** [eps1lon/dom-accessibility-api#1018]
When a div element role is `combobox` and the element has an `aria-labelledby` attribute that references itself, `computeAccessibleName` is failing to return the accessible name.

Role combobox supports name from author https://www.w3.org/TR/wai-aria/#namefromauthor

According to 2B of https://www.w3.org/TR/accname-1.2/#mapping_additional_nd_te

> if computing a name, and the current node has an aria-labelledby attribute that contains at least one valid IDREF, and the current node is not already part of an aria-labelledby traversal, process its IDREFs in the order they occur

Please correct if I misunderstand but my interpretation is it should be returning the name. I note this is also seen in Chrome dev tools in the accessibility pane.

![combobox-labelledby-self](https://github.com/eps1lon/dom-accessibility-api/assets/3656794/cee7610b-7c80-4c47-9581-fdec3b26a43d)

When role is combobox the following test cases demonstrate it failing to return an accessible name when the element refers to itself using `aria-labelledby`.

#### Test cases

```
test.each([
[
		`<div data-test id="el1" role="combobox" aria-labelledby="el1">I reference myself for my name</div>`,
		"I reference myself for my name",
	],
	[
		`
		<div data-test id="el1" role="combobox" aria-labelledby="label">I reference another element for my name</div>
		<div id="label" role="presentation">I'm prohibited a name</div>
		`,
		"I'm prohibited a name",
	],
	[
		`<div data-test id="el1" role="combobox" aria-label="I use aria-label for my name"></div>`,
		"I use aria-label for my name",
	],
])(`misc #%#`, (markup, expectedAccessibleName) => {
	expect(markup).toRenderIntoDocumentAccessibleName(expectedAccessibleName);
});
```

#### Test output

```
misc #3

    expected <div
      aria-labelledby="el1"
      data-test=""
      id="el1"
      role="combobox"
    >
      I reference myself for my name
    </div> to have accessible name 'I reference myself for my name' but got ''
    - Expected
    + Received

    - I reference myself for my name

      476 |     ],
      477 | ])(`misc #%#`, (markup, expectedAccessibleName) => {
    > 478 |     expect(markup).toRenderIntoDocumentAccessibleName(expectedAccessibleName);
          |                    ^
      479 | });
      480 |
      481 | test("text nodes are not concatenated by space", () => {

      at toRenderIntoDocumentAccessibleName (__tests__/accessible-name.js:478:17)
```

**Comment by MatanBobi** [comment]
AFAIU from the [spec](https://www.w3.org/TR/wai-aria-1.2/#combobox), that's because `role="combobox"` should only be on an `input` element and not on a `div` element.
Having said that, the `combobox` role's spec is extremely convoluted.

**Comment by agentdylan** [comment]
Agreed it is quite difficult to make sense of. I've interpreted those descriptions in the more abstract sense of input as in emulating controls rather than literal html input elements. The word input there links to https://www.w3.org/TR/wai-aria-1.2/#input describing it as 
> A generic type of [widget](https://www.w3.org/TR/wai-aria-1.2/#dfn-widget) that allows user input.

To me it appears there is intent for it to be used on div elements and we can see that in w3.org own examples e.g.
https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/

In popular UI libraries it is also common to see combobox role on div elements.
e.g. Material UI
https://mui.com/material-ui/react-select/

**Comment by MatanBobi** [comment]
I understand. I know that @eps1lon is also involved with mui so let's wait for his call on this one :)

**Comment by neaumusic** [comment]
@agentdylan Just my two cents here, but I've noticed this fits a pattern of issues I've seen; I have two WIP pull requests up that simply check a string is not empty before returning. The entire Tooltip step 2I would never have run without checking for it's predecessor. My thoughts are maybe the pattern has to do with an attempt at 2J, but who knows
